pennylane/drawer/tape_text.py

```diff
@@ -14,16 +14,23 @@
 """
 This module contains logic for the text based circuit drawer through the ``tape_text`` function.
 """
+# TODO: Fix the latter two pylint warnings
+# pylint: disable=too-many-arguments, too-many-branches, too-many-statements
 
 import pennylane as qml
-from pennylane.measurements import Expectation, Probability, Sample, Variance, State
+from pennylane.measurements import Expectation, Probability, Sample, Variance, State, MidMeasureMP
+from pennylane.tape import QuantumScript
 
 from .drawable_layers import drawable_layers
 from .utils import convert_wire_order, unwrap_controls
 
 
-def _add_grouping_symbols(op, layer_str, wire_map):
+def _add_grouping_symbols(op, layer_str, wire_map, bit_map):
     """Adds symbols indicating the extent of a given object."""
+
+    if isinstance(op, MidMeasureMP):
+        return _add_mid_measure_grouping_symbols(op, layer_str, wire_map, bit_map)
```

Comment:
If I'm reading this right, we could make `_add_mid_measure` call `_add_mid_measure_grouping_symbols` directly instead of `_add_grouping_symbols`.  Then we could get rid of this dispatch here.